detect_os

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+UNAME_PATTERNS=( "\.el[0-9]+" "Ubuntu" )
+UNAME_MATCHES=( "rhel" "ubuntu" ) #always use lowercase
+
+usage()
+{
+    echo -e "detect_os"
+    echo -e "\t Detects the OS that the system is currently using and prints it out"
+    echo -e "Options"
+    
+    echo -e "\t --uname"
+    echo -e "\t\t Determine the current OS using 'uname -a' output"
+    echo -e "\t\t Combining this option with --os-release produces undefined behavior"
+
+    echo -e "\t --os-release"
+    echo -e "\t\t Determine the current OS using the /etc/os-release file"
+    echo -e "\t\t Combining this option with --uname produces undefined behavior"
+
+    echo -e "\t -h/--help/--usage"
+    echo -e "\t\t Displays this message"
+
+    exit 0
+}
+
+parse_os_release_file()
+{
+    local os_id=`grep '^ID=' /etc/os-release | cut -d'=' -f2 | sed -e "s/\"\|'//g`
+
+    echo $os_id
+}
+
+parse_uname()
+{
+    local os="Unknown"
+    local kinfo=`uname -a`
+
+    for idx in "${!UNAME_PATTERNS[@]}"; do
+        echo $kinfo | grep -Eq "${UNAME_PATTERNS[$idx]}"
+
+        if [ $? -eq 0 ];then
+            os="${UNAME_MATCHES[$idx]}"
+        fi
+    done
+    echo $os
```

Comment:
Need to catch unknown OS's.  A message stating that, and what to update to fix the issue.